snapcraft/file_utils.py

```diff
@@ -139,16 +139,19 @@ def create_similar_directory(source, destination, follow_symlinks=False):
     :param bool follow_symlinks: Whether or not symlinks should be followed.
     """
 
-    stat = os.stat(source, follow_symlinks=follow_symlinks)
-    uid = stat.st_uid
-    gid = stat.st_gid
-    os.makedirs(destination, exist_ok=True)
-    try:
-        os.chown(destination, uid, gid, follow_symlinks=follow_symlinks)
-    except PermissionError as exception:
-        logger.debug('Unable to chown {}: {}'.format(destination, exception))
+    if os.path.exists(source):
```

Comment:
You didn't answer my question here: why would `create_similar_directory()` be called if the directory we want to imitate (`source`) doesn't exist?